app-server/database.js

```diff
@@ -0,0 +1,771 @@
+
+    var u = new User();
+    u.id = user.id;
+    u.email = user.email;
+    u.accessLevel = user.access_level;
+    return u;
+  }
+}
+
+
+
+
+
+/*
+ * Producing Organizations
+ */
+
+class ProducingOrg {
+  static getAll() {
+    return new Promise((resolve, reject) => {
+      var connection = getMysqlConnection();
+      connection.query("select * from producing_org order by name", [], (err, results) => {
+        if (err) {
+          reject(err);
+        } else {
+          if (results.length) {
+            resolve(results.map(ProducingOrg._producingOrgRecordToObject));
+          } else {
+            resolve(null);
+          }
+        }
+        connection.destroy();
+      });
+    });
+  }
+  static getById(id) {
+    return new Promise((resolve, reject) => {
+      var connection = getMysqlConnection();
+      connection.query("select * from producing_org where id = ?", [id], (err, results) => {
+        if (err) {
+          reject(err);
+        } else {
+          if (results.length) {
+            var producingOrg = ProducingOrg._producingOrgRecordToObject(results[0]);
+            resolve(producingOrg);
+          } else {
+            resolve(null);
+          }
+        }
+        connection.destroy();
+      });
+    });
+  }
+  static getByOrgAdminUserId(userId) {
+    return new Promise((resolve, reject) => {
+      var query = `select po.* from producing_org po, user_admins_org uag
+                   where uag.user_id = ? and po.id = uag.org_id`;
+      
+      var connection = getMysqlConnection();
+      connection.query(query, [userId], (err, results) => {
+        if (err) {
+          reject(err);
+        } else {
+          if (results.length) {
+            var producingOrg = ProducingOrg._producingOrgRecordToObject(results[0]);
+            resolve(producingOrg);
+          } else {
+            resolve(null);
+          }
+        }
+        connection.destroy();
+      });
+    });
+  }
+  static create(inputOrg) {
+    return new Promise((resolve, reject) => {
+      var {userId, name, missionStatement} = inputOrg;
+
+      var insertQuery = "insert into producing_org (name, mission_statement) values (?, ?)";
+
+      var connection = getMysqlConnection();
+      connection.query(insertQuery, [name, missionStatement], (insertErr, insertResult) => {
+        if (insertErr) {
+          if (insertErr.code === 'ER_DUP_ENTRY') {
+            resolve({error: insertErr});
+          } else {
+            reject(insertErr);
+          }
+          connection.destroy();
+        } else {
+          var id = insertResult.insertId;
+
+          var userPromise = ProducingOrg._createHelper_makeUserOrgAdminAndGetUser(userId, id);
+          var orgPromise = ProducingOrg.getById(id);
+
+          Promise.all([userPromise, orgPromise]).then(
+            ([user, org]) => resolve({ user: user, producingOrg: org })
+          ).catch(
+            reason => reject(reason)
+          );
```

Comment:
If this is outside the `new Promise()` constructor, you can just return `Promise.all([])`.

```javascript
queryConnection(connection, insertQuery, [name, missionStatement])
  .then(/* helper that handles duplicates */)
  .then(orgId => {
    return Promise.all([
      ProducingOrg._createHelper_makeUserOrgAdminAndGetUser(userId, orgId),
      ProducingOrg.getById(orgId);
    ]);
  })
  .finally(connection => connection.destroy());
```